index.html

```diff
@@ -0,0 +1,464 @@
+        </p>
+        <ol class="algorithm">
+          <li>If |json_launch_handler|["route_to"] does not [=map/exist=],
+          return.
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is
+          [=list=]:
+            <ol class="algorithm">
+              <li>[=list/For each=] |entry| of |json_launch_handler|["route_to"]:
+                <ol class="algorithm">
+                  <li>If the type of |entry| is not [=string=], continue.
+                  </li>
+                  <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |entry| and return.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is not [=string=], return.
+          </li>
+          <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |json_launch_handler|["route_to"].
+          </li>
+        </ol>
+        <p class="note">
+          `route_to` accepts a list of strings as a future proofing mechanism. If a new [=route to target=] is added in the future but has not yet been implemented by all user agents, web developers can cater to the old user agents by providing fallback [=route to target=] values in a list format. User agents will use the first value that is valid to them.
+        </p>
+        <p class="note">
+          For example: if "some-new-behaviour" is added in the future a site developer can specify "[=route to/existing-client-navigate=]" to be the fallback with the following:
+        </p>
+        <pre class="note json">
+          {
+            "launch_handler": {
+              "route_to": ["some-new-behaviour", "existing-client-navigate"]
+            }
+          }
+        </pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Handling Web App Launches
+      </h2>
+      <section>
+        <h2>
+          Launching a Web App
+        </h2>
+        <p>
+          When launching a web app via any means (e.g. OS launch, protocol handling, file handling, share target, link capturing, etc.) the trigger of the launch should run the steps to [=launch a web app=] with a {{LaunchParams}} containing relevant launch information (e.g. share data in the event of a share target launch).
+        </p>
+        <p>
+          The steps to <dfn>launch a web app</dfn> are given by the following algorithm and takes a {{LaunchParams}} |params:LaunchParams|.
+        </p>
+        <ol class="algorithm">
+          <li>If |params| is null, set |params| to a new {{LaunchParams}} with {{LaunchParams/targetURL}} set to [=manifest/start_url=].
+          </li>
+          <li>Set |client| to the result of running the steps to [=prepare a web app launch client=] passing [=manifest/route_to=] and |params|.{{LaunchParams/targetURL}}.
```

Comment:
You can't really just say "passing `route_to`" since you're implicitly fetching it out of the manifest.

Make steps to launch a web app "takes a [processed manifest](https://www.w3.org/TR/appmanifest/#dfn-processed-manifest) _manifest_" in addition to the LaunchParams. (Even this is a little hand-wavy -- how does the user agent decide _which_ processed manifest to choose? For now ... leave that implied. At least we'll start our formalities with taking a processed manifest.)

Then, you could extract `launch_handler.route_to` out of the manifest. But actually, for reasons I give below, we want to pass the whole manifest down into "prepare a web app launch client", so do that.

Reply:
Done.